File: src/main/java/akostenko/aicars/keyboard/IsKeyDownListener.java
package akostenko.aicars.keyboard;

import org.newdawn.slick.Input;
import org.newdawn.slick.KeyListener;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class IsKeyDownListener implements KeyListener {

    private final Logger logger = LoggerFactory.getLogger(this.getClass());

    private final int key;
    private boolean isDown = false;

    public IsKeyDownListener(int key) {
        this.key = key;
    }

    @Override
    public void keyPressed(int key, char c) {
        if (this.key == key) {
            isDown = true;
            logger.debug("Key is down {}", key);
        }
    }

    @Override
    public void keyReleased(int key, char c) {
        if (this.key == key) {
            isDown = false;
            logger.debug("Key is up {}", key);
        }
    }

    @Override
    public void setInput(Input input) {

    }

    @Override
    public boolean isAcceptingInput() {
        return true;
    }

    @Override
    public void inputEnded() {

    }

    @Override
    public void inputStarted() {

    }

    public boolean isDown() {
        return isDown;
    }
}

File: src/main/java/akostenko/aicars/keyboard/IsKeyDownListener.kt
package akostenko.aicars.keyboard

import org.newdawn.slick.Input
import org.newdawn.slick.KeyListener
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class IsKeyDownListener(private val key: Int) : KeyListener {

    private val logger = LoggerFactory.getLogger(this.javaClass)
    var isDown = false
        private set

    override fun keyPressed(key: Int, c: Char) {
        if (this.key == key) {
            isDown = true
            logger.debug("Key is down {}", key)
        }
    }

    override fun keyReleased(key: Int, c: Char) {
        if (this.key == key) {
            isDown = false
            logger.debug("Key is up {}", key)
        }
    }

    override fun setInput(input: Input) {

    }

    override fun isAcceptingInput(): Boolean {
        return true
    }

    override fun inputEnded() {

    }

    override fun inputStarted() {

    }
}
